### DISCUSSION TITLE
关于膜电位符号的几个问题

### DISCUSSION THREAD

**Discussion by Toadally-Awesome** [fangwei123456/SpikingFlow#251] [Q&A] [answered]
您好，我将这样一个模型在MNIST上训练之后查看了一下dynamic_conv_1层的膜电位，发现其中存在负值。
`class CSNN_LIF(nn.Module):
    def __init__(self, T, print_params = True):
        super().__init__()
        self.T = T

        self.static_conv = nn.Sequential(
            nn.Conv2d(1, 8, kernel_size=3, padding=1, bias=False),
        )

        self.dynamic_conv_1 = nn.Sequential(
            neuron.LIFNode(tau = 10., v_reset = 0., decay_input = False, surrogate_function=surrogate.ATan()),
            nn.AvgPool2d(2, 2),  # 14 * 14
        )

        self.dynamic_conv_2 = nn.Sequential(
            nn.Conv2d(8, 16, kernel_size=3, padding=1, bias=False),
            neuron.LIFNode(tau = 10., v_reset = 0., decay_input = False, surrogate_function=surrogate.ATan()),
            nn.AvgPool2d(2, 2)  # 7 * 7
        )

        self.classifier = nn.Sequential(
            nn.Flatten(),
            nn.Linear(16 * 7 * 7, 8 * 4 * 4, bias=False),
            neuron.LIFNode(tau = 10., v_reset = 0., decay_input = False, surrogate_function=surrogate.ATan()),
            nn.Linear(8 * 4 * 4, 10, bias=False),
            neuron.LIFNode(tau = 10., v_reset = 0., decay_input = False, surrogate_function=surrogate.ATan()),
        )
    def forward(self, x):
      x = self.static_conv(x)

      out_spikes_counter = self.classifier(self.dynamic_conv_2(self.dynamic_conv_1(x)))
      for t in range(1, self.T):
        out_spikes_counter += self.classifier(self.dynamic_conv_2(self.dynamic_conv_1(x)))
      return out_spikes_counter / self.T`
因此产生了几个问题：
1. 负膜电位是spikingjelly默认存在的，还是我读取到的膜电位有误？
2. 若膜电位可以为负，在fire和reset时有没有对负膜电位作特殊规定（例如：不发射脉冲且重置为0），还是直接保留负膜电位到下一个time step。
3. 负膜电位若被保留，是否也有衰减行为（随着乘衰减系数而逐渐向0拉回）？
*这个代码里使用的是AvgPool，这是不是会导致下一层输入不再是0或1的二元脉冲，实际上使用Maxpool更符合SNN的设计？

**Comment by fangwei123456** [comment, marked as answer]
1.负值是默认存在的，不做限制
2.不对膜电位做特殊规定
3.衰减也不会做特殊规定。平均池化可以合并到下一层的卷积，所以不影响推理
